lib/rules/force-types-on-object-props.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * @author Przemysław Jan Beigert
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+/**
+ * @typedef {import('../utils').ComponentProp} ComponentProp
+ */
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Check if all keys and values from second object are resent in first object
+ *
+ * @param {{ [key: string]: any }} a object to
+ * @param {{ [key: string]: any }} b The string to escape.
+ * @returns {boolean} Returns the escaped string.
+ */
+const isLooksLike = (a, b) =>
+  a &&
+  b &&
+  Object.keys(b).every((bKey) => {
+    const bVal = b[bKey]
+    const aVal = a[bKey]
+    if (typeof bVal === 'function') {
+      return bVal(aVal)
+    }
+    return bVal == null || /^[bns]/.test(typeof bVal)
+      ? bVal === aVal
+      : isLooksLike(aVal, bVal)
+  })
```

Comment:
The complexity of this function is quite high given that it is only used twice. Please replace both usages with explicit checks (e.g. `property.value.type === 'Identifier' && property.value.name === 'Object'`) and remove this function.